returnn/tf/network.py

```diff
@@ -1431,14 +1435,16 @@ def _check_push_flattening_to_inputs_for_layer(layer_):
 
     def _resolve_layer(layer_):
       """
+      Flattens the layer structure, removes irrelevant layers and returns next successor layer
       :param LayerBase layer_:
+      :return Next layer in succession
       :rtype: LayerBase
       """
       while True:
         if isinstance(layer_, SubnetworkLayer):
           layer_ = layer_.subnetwork.layers["output"]
           continue
-        if isinstance(layer_, CopyLayer) and len(layer_.sources) == 1:
+        if isinstance(layer_, CopyLayer) and len(layer_.sources) == 1 and not isinstance(layer_, CastLayer):
```

Comment:
I think instead of having such explicit check for `CastLayer`, we better should do just this:
```suggestion
        if type(layer_) is CopyLayer and len(layer_.sources) == 1:
```